File: Week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{

// prompt the text from the user

    string text = get_string("Text: ");

    int total_letters = count_letters(text);
    int total_words = count_words(text);
    int total_sentences = count_sentences(text);


    double L = total_letters / (float)total_words * 100;
    double S = total_sentences / (float)total_words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

// Functions---------------------

// Calculate de number of letters in the text knowing a string is an array
int count_letters(string text)
{
    int total_letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]) != 0)
        {
            total_letters ++;
        }
    }
    // printf("%i letters\n", total_letters);
    return total_letters;
}

// Calculate de number of words in the text ie: count the number of space +1
int count_words(string text)
{
    int total_words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            total_words ++;
        }
    }
    // printf("%i words\n", total_words);
    return total_words;
}

// calculate the number of sentences in the text ie: count the number of . ! or ?
int count_sentences(string text)
{
    int total_sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            total_sentences ++;
        }
    }
    // printf("%i sentences\n", total_sentences);
    return total_sentences;
}

File: Week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

# initialize variables
total_letters = 0
total_words = 1
total_sentences = 0
exception_sentences = ['.', '!', '?']

text = input("Text: ")

# count the number of letters, words and sentences
for i in range(int(len(text))):
    if text[i].isalpha():
        total_letters += 1
    elif text[i] == ' ':
        total_words += 1
    elif text[i] in exception_sentences:
        total_sentences += 1

print(total_letters)
print(total_words)
print(total_sentences)

# apply algorithm
L = total_letters / total_words * 100
S = total_sentences / total_words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print results
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade", index)